File: src/main/java/atm/bloodworkxgaming/oeintegration/Enchantments/ModEnchantments.java
package atm.bloodworkxgaming.oeintegration.Enchantments;

import net.minecraft.enchantment.Enchantment;
import net.minecraftforge.registries.IForgeRegistry;

/**
 * Created by jonas on 19.06.2017.
 */
public class ModEnchantments {


    public static Enchantment excavationEnchantment = new ExcavationEnchantment();

    public static void registerEnchantments(IForgeRegistry<Enchantment> registry) {
        registry.register(excavationEnchantment);
    }
}

File: src/main/kotlin/atm/bloodworkxgaming/oeintegration/enchantments/ModEnchantments.kt
package atm.bloodworkxgaming.oeintegration.enchantments

import atm.bloodworkxgaming.oeintegration.OreExcavationIntegration
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Created by jonas on 19.06.2017.
 */
object ModEnchantments {
    val REGISTRY = KDeferredRegister(ForgeRegistries.ENCHANTMENTS, OreExcavationIntegration.MOD_ID)

    val ENCHANTMENT_EXCAVATION by REGISTRY.registerObject("oreexcavation") {
        ExcavationEnchantment()
    }
}
